src/components/Item.jsx

```diff
@@ -51,12 +45,53 @@ const Item = ({
     return estimatedInterval;
   };
 
+  const checkDate = (lastPurchasedDate) => {
+    if (lastPurchasedDate === null) {
+      return false;
+    } else {
+      const day = 60 * 60 * 24;
+      const now = Date.now() / 1000;
+      return now - lastPurchasedDate.seconds < day;
+    }
+  };
+
   const checked = checkDate(lastPurchasedDate);
-  const className = checked ? 'checked' : '';
+  const checkedClassName = checked ? 'checked' : '';
+
+  const groupItemClassName = () => {
+    if (
+      numberOfPurchases < 2 ||
+      (lastPurchasedDate != null &&
+        calculateLatestInterval() >= 2 * nextPurchase)
+    ) {
+      return 'inactive';
+    }
+    if (nextPurchase < 7) {
+      return 'soon';
+    } else if (nextPurchase >= 7 && nextPurchase <= 30) {
+      return 'kind-of-soon';
+    } else if (nextPurchase > 30) {
+      return 'not-so-soon';
+    }
+  };
```

Comment:
This meets the item state requirements for all the ACs, it's also super clear to read 😊